### PR TITLE
implement darkmode & various improvements

### PR DESCRIPTION
**Pull request by betinho89**
The code base was improved, and the darkmode was enabled.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,7 @@ yarn-error.log*
 
 # vercel
 .vercel
+
+# locks
+yarn.lock
+package-lock.json
```
Comment (nionis) [inline .gitignore:38]: why are we removing the `lock` files here?
It's considered best practise to include lock files, this ensures that people installing the repo will get precisely the same dependencies as you have.

**Review by nionis — CHANGES_REQUESTED** [review]

**Comment by nionis** [comment]
some feedback:

- `lock files`: https://github.com/hoprnet/hopr-leaderboard/pull/23#discussion_r514133154

- `copy to clipboard`: It doesn't always show the [tooltip](https://github.com/hoprnet/hopr-leaderboard/pull/23/files#diff-932eaa5789fa4d09a7b899a41bb4fe94bf524751aebb08dfc8a2b454bb02d7f8R26)

- some text is too dark, for example:
image
let's make them "brighter"

- I see that we use 2 different highlight colors:
   1. `#2196f3` for the `dark mode toggle`
   2. `#FFC0CB` for icons, some numbers
let's switch into using `#2196f3` so it's more consistent

File: .gitignore (hunk 1 above)
Comment (jjperezaguinaga) [reply .gitignore:38]: @nionis just as a background here, I think there was a confusion from https://github.com/hoprnet/hoprnet/pull/414 where I asked @betinho89 to remove `package-lock.json` and `yarn.lock` got included.

@betinho89 let's move `yarn.lock` outside of the `.gitignore`

**Comment by nionis** [comment]
hey @betinho89, not sure if you are done & ready for review, but just in case I've got feedback for the latest commit: https://hopr-leaderboard-iklhzio5x.vercel.app/

- rename `copy to clipboard` to `copied to clipboard`
- `copy to clipboard` background color doesn't look very nice: maybe make the font color white? feel free to try out something
copy
- highlight colours still inconsistent, let's stick into using one, I prefer blue
highlight
- mouse hover on `id` column causes the horizontal scroll to popup on a `1920x1080` screen, it's fine if it does it for smaller screens `<720p`
mouse_over
- make `id` column copy-able
- let's add to all copy-able texts the "copy" icon
- let's update [hopr-allocation](https://hopr-leaderboard-iklhzio5x.vercel.app/hopr-allocation) to https://github.com/hoprnet/hopr-leaderboard/issues/26

**Comment by nionis** [comment]
@betinho89 is this PR ready for review?

**Comment by jjperezaguinaga** [comment]
@nionis merging as its, and getting the fixes in a separate PR.

**Pull request merged**